model/simulationtests/coilsystem_cooling_water.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+require 'openstudio'
+require_relative 'lib/baseline_model'
+
+m = BaselineModel.new
+
+# make a 1 story, 100m X 50m, 1 zone core/perimeter building
+m.add_geometry({ 'length' => 100,
+                 'width' => 50,
+                 'num_floors' => 1,
+                 'floor_to_floor_height' => 4,
+                 'plenum_height' => 1,
+                 'perimeter_zone_depth' => 0 })
+
+# add windows at a 40% window-to-wall ratio
+m.add_windows({ 'wwr' => 0.4,
+                'offset' => 1,
+                'application_type' => 'Above Floor' })
+
+# add thermostats
+m.add_thermostats({ 'heating_setpoint' => 24,
+                    'cooling_setpoint' => 28 })
+
+# assign constructions from a local library to the walls/windows/etc. in the model
+m.set_constructions
+
+# set whole building space type; simplified 90.1-2004 Large Office Whole Building
+m.set_space_type
+
+# add design days to the model (Chicago)
+m.add_design_days
+
+# add ASHRAE System type 07, VAV w/ Reheat
+m.add_hvac({ 'ashrae_sys_num' => '07' })
+
+# In order to produce more consistent results between different runs,
+# we sort the zones by names (only one here anyways...)
+zones = m.getThermalZones.sort_by { |z| z.name.to_s }
+
+### Water Size Economizer Model ###
+
+zone = zones[0]
+airloop = zone.airLoopHVAC.get
+airloop.setName('AirLoopHVAC CoilSystemCoolingWater')
+
+# create a CoilSystemCoolingWater object
+coil_system = OpenStudio::Model::CoilSystemCoolingWater.new(m)
+coil_system.setName('CoilSystemCoolingWater WaterSize')
+
+# Replace the default CoilCoolingWater with CoilSystem, then remove the default one
+# Note: it's probably going to end up badly controlled when you place the
+# CoilSystem directly on the AirLoopHVAC path and not inside another component
+# (such as UnitarySys, cf: https://github.com/NREL/EnergyPlus/issues/7222)
+water_coil = coil_system.coolingCoil.to_CoilCoolingWater.get
+water_coil.setName('CoilSystemCoolingWater WaterSize CoolingCoil')
+
+coil = airloop.supplyComponents(OpenStudio::Model::CoilCoolingWater.iddObjectType).first.to_CoilCoolingWater.get
+# Note that we connect the CoilSystem, NOT the underlying CoilCoolingWater
+coil_system.addToNode(coil.airOutletModelObject.get.to_Node.get)
+plant = coil.plantLoop.get
+# But we have to connect the water_coil itself...
+plant.addDemandBranchForComponent(water_coil)
+coil.remove
```

Comment:
TODO: We need a packaged DX system or coil cooling water downstream of the CoilSystem here on the AirLoopHVAC, but you're removing it.

The coil should we connected to a CondenserLoop on the demand side I think ideally, here it's a chilled water loop

> As a water-side economizer the [CoilSystem:Cooling:Water](https://bigladdersoftware.com/epx/docs/24-2/input-output-reference/group-heating-and-cooling-coils.html#coilsystemcoolingwater) object is placed upstream of packaged DX systems or chilled water main cooling coil